via/static/scripts/video_player/components/CopyButton.tsx

```diff
@@ -1,25 +1,32 @@
 import { CopyIcon, IconButton } from '@hypothesis/frontend-shared';
 
+import type { ToastMessageAppender } from '../hooks/use-toast-messages';
 import { formatTranscript } from '../utils/transcript';
 import type { TranscriptData } from '../utils/transcript';
 
 export type CopyButtonProps = {
   transcript: TranscriptData | null;
+  appendToastMessage: ToastMessageAppender;
 };
 
 /**
  * Toolbar button that copies the transcript to the clipboard.
  */
-export default function CopyButton({ transcript }: CopyButtonProps) {
+export default function CopyButton({
+  transcript,
+  appendToastMessage,
+}: CopyButtonProps) {
   const copyTranscript = async () => {
     const formattedTranscript = transcript
       ? formatTranscript(transcript.segments)
       : '';
     try {
       await navigator.clipboard.writeText(formattedTranscript);
     } catch (err) {
-      // TODO: Replace this with a toast message in the UI.
-      console.warn('Failed to copy transcript', err);
+      appendToastMessage({
+        type: 'error',
+        message: 'Failed to copy transcript',
+      });
```

Comment:
I would reconsider displaying a toast message when the copy succeeds. Currently it feels like nothing happened.

Reply:
I agree, especially since the button itself doesn't provide any kind of visual information. A green "Transcript copied" message would be fine.